tests/test_runlib.py

```diff
@@ -501,6 +503,47 @@ def test_in_toto_run_compare_dumped_with_returned_link(self):
         FILENAME_FORMAT.format(step_name=self.step_name, keyid=self.key["keyid"]))
     self.assertEqual(repr(link), repr(link_dump))
 
+  def test_in_toto_run_with_metadata_directory(self):
+    """Successfully run with metadata directory,
+     compare dumped link is equal to returned link"""
+    try:
+      tmp_dir = tempfile.mkdtemp()
+      link = in_toto_run(self.step_name, [self.test_artifact],
+          [self.test_artifact], ["python", "--version"], True, self.key,
+          metadata_directory=tmp_dir)
+      file_path = os.path.join(tmp_dir, self.step_name)
+      link_dump = Metablock.load(
+          FILENAME_FORMAT.format(step_name=file_path, keyid=self.key["keyid"]))
+      self.assertEqual(repr(link), repr(link_dump))
+
+    #clean up
+    finally:
```

Comment:
I still think we should reorganize these, perhaps all under a class of tests for metadata directory, and use `tearDown` / `tearDownClass` for clean up. We shouldn't need `try` here.

Reply:
Though, I think it'd be a first to have a class that's not for a specific method defined in `runlib`, so we may not want to go that direction either.